python/vyos/ifconfig/interface.py

```diff
@@ -336,8 +366,26 @@ def __init__(self, ifname, **kargs):
         self.operational = self.OperationalClass(ifname)
         self.vrrp = VRRP(ifname)
 
+    def get_interface_within_netns(self, key):
+        if self.netns:
+            return f"ip netns exec {self.netns} {self._command_get[key]['shellcmd'].format(**self.config)}"
+        else:
+            return self._command_get[key]['shellcmd'].format(**self.config)
+
+    def set_interface_within_netns(self, key, value):
+        if self.netns:
+            cmd = f"ip netns exec {self.netns} {self._command_set[key]['shellcmd'].format(**self.config, value=value)}"
+        else:
+            cmd = self._command_set[key]['shellcmd'].format(**self.config, value=value)
+        self._cmd(cmd)
+
     def _create(self):
-        cmd = 'ip link add dev {ifname} type {type}'.format(**self.config)
+        # Do not create interface that already exist or exists in netns
+        if self.exists(f'{self.ifname}') or get_interface_namespace(self.config['ifname']):
```

Comment:
Can we be smarter about this check? The `get_interface_namespace` implementation is going to slow linearly with the number of namespaces given how it's implemented (I don't really know if/think there is a better way to implement "find the namespace given the interface name" though). We know the current network namespace (or the old namespace if it was changed in the latest commit) so it  shouldn't be necessary to do the slow(er) thing here.